### DISCUSSION TITLE
[Jetstream] Multiple hierarchical Subscriptions

### DISCUSSION THREAD

**Discussion by in-cloud-opensource** [nats-io/nats.go#1002] [Q&A]
Hi,

We are building a central component that handles deletion of data in a common way. For that, we are planning to publish a single event to delete data in all subscribers.

```
// use case: delete all data in all services
Publish: data.delete
```

Apart from that we want to trigger data deletion for a single service.
```
// use case: delete data for a single service “some-service”
Publish: data.delete.some-service
```
Is there a way to configure a consumer in way to get data from `data.delete` and `data.delete.some-service` in the same time or do we have to use multiple consumers, where every service needs multiple parallel subscribe calls

We are using the nats Jetstream

Regards,

**Comment by ripienaar** [comment]
subscribe to `data.delete.*` and then publish to like `data.delete.all` or `data.delete.some-service`

**Reply by in-cloud-opensource** [reply]
But when I subscribe to `data.delete.*` I would expect that every subscriber gets all messages for every service, right? Then clients need to filter I they are interested in the message that have been published.

**Comment by ripienaar** [comment]
Yes, then you need to subscribe multiple times.  Is this JetStream or core nats?

**Reply by in-cloud-opensource** [reply]
Jetstream

**Comment by ripienaar** [comment]
Yes, then you need multiple subscribes and probably for each one an ephemeral consumers.  If you want it to act like all subscribers with servce service-name gets it each will need a consumer.
